Makefile

```diff
@@ -38,9 +38,12 @@ default: build test
 #
 
 # Dependencies for the base version (English 1.0)
-asm_files =  $(shell find src     -type f -name '*.asm' -o -name '*.inc')
-gfx_files =  $(shell find src/gfx -type f -name '*.png')
-bin_files =  $(shell find src     -type f -name '*.tilemap.encoded' -o -name '*.attrmap.encoded')
+asm_files :=  $(wildcard src/*.asm)
+asm_files +=  $(wildcard src/*.inc)
```

Comment:
As far as I understand, this is enough to compile the project - but not to track dependencies properly when rebuilding.

More specifically, this expression misses most of asm files dependencies. For instance this case works on `main`, but not on this PR:

```shell
➜ make # initial compilation
rgbasm --export-all -DLANG=EN -DVERSION=0 -I src/ -o src/main.azle.o src/main.asm
rgblink   -n azle.sym -o azle.gbc src/main.azle.o
rgbfix --color-compatible --sgb-compatible --ram-size 0x03 --old-licensee 0x33 --new-licensee "01" --mbc-type 0x1B --pad-value 0xFF --validate --rom-version 0 --non-japanese --title "ZELDA" azle.gbc
azle.gbc checksum: OK
➜ touch src/code/intro.asm # touch any asm file in `src/code`
➜ make
azle.gbc checksum: OK # nothing was rebuilt - but it should have, because dependencies changed
```

It works for building, because all `asm` files are included by `src/main.asm` : the assembler is invoked only once, with `main.asm`, so everything gets compiled anyway. But as dependencies are not properly tracked, re-building after a file changed doesn't work.

I think it may just need your variant of "recursive wildcard" again?

Reply:
Perhaps, though the “include every asm file in main” seems a bit hackish. Could be time to individually link files, but for now, I’ll do what I did for the gfx files, since nothing is more than three layers deep in subdirectories.

Reply:
addressed this concern in the most recent commit